### ISSUE TITLE
Microsoft.NET.Sdk.Functions (v1.0.33) breaks function apps that have a dependency on Microsoft.Extensions.Logging.Abstractions v2.2.0

### ISSUE THREAD

**Issue by f2bo** [Azure/azure-functions-host#5750]
After upgrading from Microsoft.NET.Sdk.Functions v1.0.31 to v1.0.33, all functions in a V1 app fail with the following error:

```
FunctionXXX: Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.
```

#### Investigative information
It seems that the problem is that the version of Microsoft.Extensions.Logging.Abstractions referenced by the application is no longer being loaded. 

Before the package upgrade, with v1.0.31, two versions of this DLL were loaded, one from the AzureFunctionsTools directory and another from the bin directory of the app. When v1.0.33 is used, only the DLL in the AzureFunctionsTools directory is loaded.

#### Repro steps
- Create a new Functions V1 project
- Change the target framework v4.72 (not sure if this is relevant, probably not)
- Install the Microsoft.Extensions.Logging.Abstractions v2.2.0 NuGet package
- **Upgrade Microsoft.NET.Sdk.Functions to v1.0.33** 
- Replace Function1 with the following:
   ``` C#
    [FunctionName("Function1")]
    public static async Task<HttpResponseMessage> Run([HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)]HttpRequestMessage req)
    {
        // reference any data type from Microsoft.Extensions.Logging, for example:
        var level = Microsoft.Extensions.Logging.LogLevel.Information;
           
        return req.CreateResponse(HttpStatusCode.OK, "Done!");
    }
   ```
- Run the application locally

#### Expected behavior

- Application should start normally

#### Actual behavior

- Function fails to load with the following errors:
```
[3/5/2020 6:30:59 PM] Function 'Function1.Run' failed indexing and will be disabled.
[3/5/2020 6:30:59 PM] No job functions found. Try making your job classes and methods public. If you're using binding extensions (e.g. ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. config.UseServiceBus(), config.UseTimers(), etc.).
[3/5/2020 6:30:59 PM] The following 2 functions are in error:
[3/5/2020 6:30:59 PM] Function1: Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.
[3/5/2020 6:30:59 PM] Run: Microsoft.Azure.WebJobs.Host: Error indexing method 'Function1.Run'. mscorlib: Could not load file or assembly 'Microsoft.Extensions.Logging.Abstractions, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The system cannot find the file specified.
```

#### Known workarounds

Revert Microsoft.NET.Sdk.Functions to v1.0.31.

UPDATED: Repro steps

**Comment by mhoeger** [comment]
Thanks so much for all the details @f2bo . Adding @fabiocav (although he is out the next couple of days)

**Comment by brettsam** [comment]
Closing as a dupe of https://github.com/Azure/azure-functions-host/issues/5747. Follow that issue for more details.

**Comment by f2bo** [comment]
@brettsam Hmm.. ok, I'm having a little difficulty seeing the similarity and how extensions.json or dependency injection is involved here, but presumably you recognize the same underlying cause.

**Comment by mhoeger** [comment]
@f2bo - per Brett:
> The big difference between 3.0.3 and 3.04 is that we trim the output now.

We're trimming too aggressively, so that's causing both the DI issue and your issue. I believe a fix is coming soon, but that can be tracked with #5747!

edit: nvm :)

**Comment by brettsam** [comment]
Oh geez, sorry, I closed the wrong issue (too many tabs open!). 

This is a Functions v1 project? Can you share the `csproj` file?

**Comment by f2bo** [comment]
No problem. 

Yes, this is V1. Here's the project file for the repro project.

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net472</TargetFramework>
    <AzureFunctionsVersion>v1</AzureFunctionsVersion>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="2.2.0" />
    <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="1.0.33" />
  </ItemGroup>
  <ItemGroup>
    <Reference Include="Microsoft.CSharp" />
  </ItemGroup>
  <ItemGroup>
    <None Update="host.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
    <None Update="local.settings.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
    </None>
  </ItemGroup>
</Project>
```

**Comment by brettsam** [comment]
If you add `<_FunctionsSkipCleanOutput>true</_FunctionsSkipCleanOutput>` as a property in your csproj does it get you working again?

**Comment by f2bo** [comment]
Yes, it does help. At least I don't see any errors when launching the repro project. Though, I'll have to test this in the real project.

Can you explain what this does? It seems to be undocumented. I just want to be clear on whether this setting is permanent or a temporary workaround until something is fixed in a future release.

Thanks!

**Comment by brettsam** [comment]
It's brand new -- our latest release trims the output of all assemblies that we already ship as part of the host. It seems like you actually want that version to stay there so it's affecting you.

I'm curious why you need Logging.Abstractions 2.2.0? Is it b/c a dependency needs it?

However, there's no real requirement for you to move to 1.0.33, either -- if 1.0.31 works for you, you can stick with it...

**Comment by f2bo** [comment]
About not moving to 1.0.33, you're correct. But this is an app that hasn't gone into production yet and we usually try to keep up with the latest releases while developing so that we don't fall too far behind and then find out at the last minute that we need to upgrade and there's some major incompatibility.

And yes, we do need Logging.Abstractions 2.2.0 because it's required by a dependency. All our libraries are currently on v2.2.0. Unfortunately, we have a dependency that's only available on .NET Framework (ADOMD Analysis Services client) and we've had to split our app into V1 and V2 functions. V1 has been very painful in this respect. For example, we cannot bind ILogger and are stuck with TraceWriter because we are loading v2.2.0 and the host is still targeting v1.1.1.0 (not sure why). 

Anyway... you asked, so allow me a little rant.  :) 

Thanks!
